draft-ietf-masque-connect-udp-listen.md

```diff
@@ -153,11 +163,114 @@ UDP Payload:
+disabled using {{restrictingips}}.
+
+## Header Compression {#compression}
+
+The client MAY choose to map IP and port information per datagram against the
+Context ID defined in {{contextid}}. In such a case, the client MUST send a
+COMPRESSION_REQUEST capsule (see {{capsulerequestformat}}) with the target
+information (see {{targetmappingformat}}) it wishes to compress and the proxy
+MUST respond with either a COMPRESSION_ASSIGN (see {{capsuleresponseformat}})
+capsule if it accepts the compression request, or a COMPRESSION_REJECT
+(see {{capsulerejectformat}}) if it doesn't wish to support compression
+(For example, due to considerable memory requirements of establishing a list
+of mappings per target per client). If the compression was rejected,
+the client and server MUST continue to use uncompressed, Context ID 0 Datagrams
+for the requested target.
+Otherwise, they must exchange compressed datagrams for the given datagram as
+described in {{mappings}}.
+
+### Compression Mapping {#mappings}
+
+If a COMPRESSION_ASSIGN is received by client, the proxy and client MUST
+maintain a compression mapping from Target to Context ID and update this
+mapping whenever new COMPRESSION_REQUEST and COMPRESSION_ASSIGN capsules
+are exchanged between the two. Upon completing this exchange, the proxy and
+the client MUST send each other packets corresponding to a target using the
+unique Context ID assigned to the given target, while omitting IP and port
+information.
+
+
+## Restricting IPs {#restrictingips}
+
+The client MAY request the proxy reject any uncompressed targets by sending
+it the COMPRESSION_CLOSE (see {{compressioncloseformat}}) capsule with the
+enable field set to true. Upon receiving the request, the proxy MUST echo
+back the COMPRESSION_CLOSE with the enable field set to true. Hereon, the
+proxy MUST not forward any packets from uncommpressed targets to the client
+and MUST drop any UDP  payloads received from unregistered targets.
+
+Likewise, the client MAY choose to re-enable traffic from uncompressed
+targets, by sending the COMPRESSION_CLOSE capsule with the enable field
+set to false and the proxy MUST respond with COMPRESSION_CLOSE with enable
+set to false and MUST re-start forwarding uncompressed context ID 0 payloads.
+
+
+## Capsules {#capsules}
+The Listener capsule types are defined below:
+
+~~~ ascii-art
+Capsule {
+  Type COMPRESSION_REQUEST (0x04),
+  Length (i),
+  Target Information,
+}
+~~~
+{: #capsulerequestformat title="Compression Request Capsule Format"}
+
+~~~ ascii-art
+Capsule {
+  Type COMPRESSION_ASSIGN (0x05),
+  Length (i),
+  Target Information,
+}
+~~~
+{: #capsuleresponseformat title="Compression Assign Capsule Format"}
+
+
+~~~ ascii-art
+Capsule {
+  Type COMPRESSION_REJECT (0x06),
+  Length (i),
+  Target Information,
+}
+~~~
+{: #capsulerejectformat title="Compression Reject Capsule Format"}
+
+
+~~~ ascii-art
+Capsule {
+  Type COMPRESSION_CLOSE (0x07),
+  enable (1),
```

Comment:
I don't think I understand what the enable bit means, can you elaborate?

Reply:
Removed it